compiler/src/main/scala/org/scalajs/nscplugin/GenJSExports.scala

```diff
@@ -354,8 +369,19 @@ trait GenJSExports[G <: Global with Singleton] extends SubComponent {
 
       val overloads = alts.map(new ExportedSymbol(_, static))
 
-      val (formalArgs, restParam, body) =
-        genOverloadDispatch(jsName, overloads, jstpe.AnyType)
+      val (formalArgs, restParam, body) = {
+        if (allowCallsiteInlineSingle && overloads.size == 1) {
```

Comment:
Maybe we should take this fast path even when `!allowCallsiteInlineSingle`, and use `inline = allowCallsiteInlineSingle` a few lines below?